modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/service/AsahSegmentsEntryLocalServiceWrapper.java

```diff
@@ -105,11 +105,38 @@ public SegmentsEntry recalculateSegmentsEntry(long segmentsEntryId)
 		Criteria criteria = expressionContext.accept(
 			new IndividualSegmentsExpressionVisitorImpl());
 
-		return updateSegmentsEntry(
+		return super.updateSegmentsEntry(
 			segmentsEntry.getSegmentsEntryId(), individualSegment.getId(),
 			nameMap, null, true, _serialize(criteria), serviceContext);
 	}
 
+	@Override
+	public SegmentsEntry updateSegmentsEntry(
+			long segmentsEntryId, String segmentsEntryKey,
+			Map<Locale, String> nameMap, Map<Locale, String> descriptionMap,
+			boolean active, String criteria, ServiceContext serviceContext)
+		throws PortalException {
+
+		SegmentsEntry segmentsEntry =
+			_segmentsEntryLocalService.getSegmentsEntry(segmentsEntryId);
+
+		IndividualSegment individualSegment =
+			_asahFaroBackendClient.getIndividualSegment(
+				segmentsEntry.getCompanyId(),
+				segmentsEntry.getSegmentsEntryKey());
+
+		individualSegment.setFilter(criteria);
+		individualSegment.setName(
+			nameMap.get(
+				_portal.getSiteDefaultLocale(
+					serviceContext.getScopeGroupId())));
+
+		_asahFaroBackendClient.updateIndividualSegment(
+			segmentsEntry.getCompanyId(), individualSegment);
+
+		return recalculateSegmentsEntry(segmentsEntryId);
```

Comment:
Why we have to retrieve the data from AC . As per my understanding, we should have the same data in both sites because before that we are updating the data to AC.